README.md

```diff
@@ -235,6 +235,137 @@ If you don't see a command prompt, try pressing enter.
 / # wget -O - <NGINX_POD_IP>
 ```
 
+# Testing
+
+## E2E testing
+
+E2E testing purpose is to validate submariner behaviour from an integration point of
+view. It needs to be executed in connection to an existing set of clusters.
+
+Config needs to be provided via a single kubeconfig file, you can squash it
+following the next command:
+
+  ```bash
+  export KUBECONFIG=creds/cluster1:creds/cluster2:creds/cluster3
+  kubectl config view --merge=true --flatten > creds/clusters-123
+  export KUBECONFIG=creds/clusters-123
+  ```
+
+E2E tests will identify each separate cluster by the config contexts, and
+we need to tell e2e tests which contexts to use by passing the -dp-context
+flag several times. (-dp-context means "Dataplane Context")
+
+Assuming that we have cluster1, cluster2 and cluster3 contexts, and that
+cluster1 is our broker cluster with no dataplane connectivity provided by
+submariner, **to execute the E2E tests** we would do:
+
+  ```bash
+  cd test/e2e
+  go test -args -kubeconfig=creds/clusters-123 \
+                -dp-context cluster2 \
+                -dp-context cluster3 \
+                -ginkgo.randomizeAllSpecs
+  ```
+
+If you want to execute just a subset of the available E2E tests, you can use:
```

Comment:
"...,  you can specify the ginkgo.focus argument"

Reply:
Thanks, done.